app/controllers/survey.rb

```diff
@@ -0,0 +1,49 @@
+# get '/' do
+#    redirect '/surveys'
+# end
+
+get '/surveys' do
+  # @surveys = Survey.all
+  erb :index
+end
+
+get '/surveys/new' do
+  erb :'/surveys/new'
+end
+
+post '/surveys' do
+  @survey = Survey.create(params['survey'])
+  redirect "/surveys/#{@survey.id}/questions/new"
+end
+
+
+get '/surveys/:id' do
+  @questions = Question.where(survey_id: params[:id])
+  # if Answer.exists?(survey_id: params[:id])
+  if Answer.find_by(survey_id: params[:id]).user_id == current_user.id
```

Comment:
This work, that is, to work out whether the current user has answered any questions in this survey is either the user's job or the survey's job.

```ruby
survey = Survey.find(params[:id])
if survey.user_has_completed?(current_user)
# ...
# or on the user model:
if current_user.has_completed_survey?(survey)
# ...
```